### PR TITLE
ci: exit with none zero code when snapshots updated (only in github actions CI env).

### LINKED ISSUE #2005: [Bug]: should exit none 0 when the snapshots updated.
### Reproduction link or steps

1. try to update some code in core, that will affect codegen.
2. run `cargo test` 
3. echo the exit code

### What is expected?

should exit none 0 when the snapshots updated.

### What is actually happening?

exit  0 when the snapshots updated.

### System Info

```Shell
n/a
```

### Any additional comments?

_No response_

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
### Description
Closed https://github.com/rolldown/rolldown/issues/2005

### CODE REVIEW

File: crates/rolldown/tests/integration_rolldown.rs [hunk 1]
```diff
@@ -61,5 +62,15 @@ async fn filename_with_hash() {
...
+  let snapshot_path = workspace::crate_dir("rolldown")
+    .join("tests/snapshots/integration_rolldown__filename_with_hash.snap");
+  let before_snapshot = std::fs::read_to_string(&snapshot_path).unwrap();
+
   insta::assert_snapshot!(snapshot_outputs.join("\n"));
+
+  let in_github_action = std::env::var("GITHUB_ACTIONS").is_ok();
+  let after_snapshot = std::fs::read_to_string(&snapshot_path).unwrap();
+  if (before_snapshot != after_snapshot && in_github_action) {
+    std::process::exit(-1);
```
Comment (hyf0) [inline crates/rolldown/tests/integration_rolldown.rs]: Should  this be done by insta?

https://github.com/rolldown/rolldown/blob/dde9b33b409bfc909d089769d2817f178e00f30f/.github/workflows/reusable-cargo-test.yml#L36-L37

Comment (IWANABETHATGUY) [reply crates/rolldown/tests/integration_rolldown.rs]: you are right.

Comment (IWANABETHATGUY) [reply crates/rolldown/tests/integration_rolldown.rs]: https://github.com/rolldown/rolldown/actions/runs/10431525166/job/28891291395?pr=2006, but actually it did not work

Comment (IWANABETHATGUY) [reply crates/rolldown/tests/integration_rolldown.rs]: image

Comment (IWANABETHATGUY) [reply crates/rolldown/tests/integration_rolldown.rs]: I agree with you that setting the env is correct way to fix this issue.

**Review by hyf0 — APPROVED** [review]

**Comment by IWANABETHATGUY** [comment]
The ci failed because Windows error message is different from linux and mac, 
image

**Comment by hyf0** [comment]
Let's ignore this test temparaly.

**Pull request merged**